shared-libs/transitions/src/transitions/index.js

```diff
@@ -161,8 +136,9 @@ const loadTransitions = () => {
     const conf = transitionsConfig[transition];
 
     const disabled = conf && conf.disable;
+    const allowSync = conf && !conf.async;
 
-    if (!conf || disabled) {
+    if (!conf || disabled || (!allowSync && synchronous)) {
```

Comment:
I chose to make the config "disable" some transitions from running synchronously instead of "enabling" to run synchronously (after back and fourth-ing a few times). Reasoning was to enable projects to start using this feature without changes to configurations.
AFAICT the only transitions which **should** run async are `muting` and `update_notifications`, which may end up needing to updated lots of documents.

I would appreciate some feedback on this.

Reply:
I'm wondering if we should just hardcode those ones as async and not allow configuration override. Something like...

```js
module.exports = {
  filter: () => {...},
  onMatch: () => {...},
  asynchronousOnly: true // defaults to false
};
```

We definitely want to default to sync for the rest of the transitions. The most important bits to sync are the validations and replies.

Reply:
Ok, great! Will do.

Reply:
I'm not sure I agree that "any transition that is sync safe can be run in parallel / in batches", but maybe I don't understand what you mean...

> I'm pretty sure that multi_report_alerts need to be only run async, because it counts reports written to disk, and so if done in a batch you could miss some.

I don't think sync exec has anything to do with batching. I think it's fine to run it sync but not batched. The only downside is it's quite a slow transition so it'll slow down the response massively. It's also not that useful to do in sync because it usually doesn't respond to the sender but some supervisor.

Reply:
Updated so transitions are forced to by async only by hardcoded `asynchronousOnly`.
`muting`, `update_notifications` and `multi_report_alerts` run async only.

Reply:
> I'm not sure I agree that "any transition that is sync safe can be run in parallel / in batches"

We can wax on at standup about what this kind of things means, it's kind of hard to do in PR comments. It's also kind of confusing because sync and async are overloaded here to mean potentially opposite things (e.g. when I say sync do I mean sync as part of SMS writing, or sync in that they run one after the other in sentinel, and visa versa for async).

What I meant there though, is that this is a true and required property of running a transition before writing the SMS by definition of this PR (unless I misread it, in which case I apologise!, or it changes), because in the PR SMS **are** being run through in a batch, so it has to be part of the decision making on which transitions are OK to run here[1]. I can see your additional point about performance (not bothering to run transitions sync that aren't going to potentially generate responses), which is a great thought here as well which I did not consider.

[1] e.g. you have a multi-report-alert configured to send an SMS if you see three malaria reports in 24 hours, there is already one on disk, you get two in the batch of SMS from gateway: if this transition ran at SMS-write then both reports would think there are two malaria reports, as opposed to the first thinking there is two and the second thinking there is three.

Reply:
Hmm, I think this is indeed an issue with other transitions as well. 
Fe: `accept_patient_reports` with clearing schedules. 

Will have to update so docs are saved after transitions run over them, in sequence versus in batch.